src/RoboSaverVirtualModule.sol

```diff
@@ -268,15 +295,28 @@ contract RoboSaverVirtualModule {
         /// @dev Last argument `1` stands for `OperationType.DelegateCall`
         delayModule.execTransactionFromModule(MULTICALL3, 0, multicallPayload, 1);
 
-        emit AdjustPoolTxDataQueued(MULTICALL3, abi.encode(calls_));
+        uint256 cachedQueueNonce = delayModule.queueNonce();
+        txQueueData = TxQueueData(cachedQueueNonce, MULTICALL3, multicallPayload);
+
+        emit AdjustPoolTxDataQueued(MULTICALL3, abi.encode(calls_), cachedQueueNonce);
         emit PoolDepositQueued(_card, _surplus, block.timestamp);
 
         return calls_;
     }
 
-    /// @notice Check if there is a transaction queued up in the delay module
+    /// @dev Execute the next transaction in the queue using the storage variable `txQueueData`
+    function _executeNextTx() internal {
+        address cachedTarget = txQueueData.target;
+        delayModule.executeNextTx(cachedTarget, 0, txQueueData.payload, cachedTarget == MULTICALL3 ? 1 : 0);
+
```

Comment:
maybe an event here, to log successful execution?

Reply:
320bfcf269c8a72cc02d7f1c64d24eadf43f49c7